### ISSUE TITLE
ShardingPreparedStatement Class exception not caught

### ISSUE THREAD

**Issue by jimmycheng888** [apache/incubator-shardingsphere#2302]
**Which version of ShardingSphere did you use?**
3.1.0

**Which project did you use? Sharding-JDBC or Sharding-Proxy?**
Sharding-JDBC

**Actual behavior**
The code exception is not caught, causing a null pointer exception, and does not throw a true exception, making it impossible for the developer to accurately determine what went wrong.

**Example codes for reproduce this issue (such as a github link).**
The code  packge is : io.shardingsphere.shardingjdbc.jdbc.core.statement

For example, the following code, the special symbol will be a direct null pointer, and the real reason is not exposed.

   ```
 @Override
    public boolean execute() throws SQLException {
        try {
            clearPrevious();
            sqlRoute();
            initPreparedStatementExecutor();
            return preparedStatementExecutor.execute();
        } finally {
            refreshTableMetaData(connection.getShardingContext(), routeResult.getSqlStatement());
            clearBatch();
        }
    }
```

**Comment by DongJigong** [comment]
I think this code block throws an SQLException. they never catch it. please reconfirm

**Comment by terrymanu** [comment]
Can you provide exception info?

**Comment by cherrylzhao** [comment]
hi, we have fixed it in 4.0.0-RC1, you can see #1878

**Comment by jimmycheng888** [comment]
> Can you provide exception info?


Lùrù fēifǎ SQL yǔjù jiù néng chóng xiàn, bǐrú ShardingSphere bù zhīchí jiěxī de SQL biāozhì fú jiù néng chóng xiàn, huòzhě hint fēn piàn shí, rènyì pāo chū yìcháng yě kěyǐ chóng xiàn zhège bug
您是不是要找： 输入非法SQL语句就能重现,比如Sharding Sphere 不支持解析的SQL标识符就能重现,或者hint分片时,任意抛出异常也可以重现这个bug
74/5000
Entering an illegal SQL statement can be reproduced. For example, ShardingSphere does not support parsing SQL identifiers to reproduce, or when sharding, any exception can be thrown to reproduce the bug.
